client/driver/fabfile.py

```diff
@@ -1029,3 +1029,93 @@ def task_status_ui_test():
+        summary = json.load(fin)
+        summary['start_time'] = i * 20000000
+        summary['observation_time'] = 100
+        summary['end_time'] = (i + 1) * 20000000
+    with open('./integrationTests/data/x__summary.json', 'w') as fout:
+        json.dump(summary, fout)
+    return gain
+
+
+@task
+def integration_tests():
+    # Create test website
+    response = requests.get(dconf.WEBSITE_URL + '/test/create/')
+    LOG.info(get_content(response))
+
+    # Upload training data
+    LOG.info('Upload training data to no tuning session')
+    upload_batch(result_dir='./integrationTests/data/', upload_code='ottertuneTestNoTuning')
+
+    # periodic tasks haven't ran, lhs result returns.
+    LOG.info('Test no pipeline data, LHS returned')
+    upload_result(result_dir='./integrationTests/data/', prefix='0__',
+                  upload_code='ottertuneTestTuningGPR')
+    response = get_result(upload_code='ottertuneTestTuningGPR')
+    assert response['status'] == 'lhs'
+
+    # wait celery periodic task finishes
+    assert wait_pipeline_data_ready(), "Pipeline data failed"
+
+    total_n = 30
+    first_n = 5
+    last_n = 5
+    average = 0
+    simulate_db_run(1, {'effective_cache_size': '0kB'})
+    for i in range(2, total_n + 2):
+        LOG.info('Test GPR (gaussian process regression)')
+        upload_result(result_dir='./integrationTests/data/', prefix='x__',
+                      upload_code='ottertuneTestTuningGPR')
+        response = get_result(upload_code='ottertuneTestTuningGPR')
+        assert response['status'] == 'good'
+        gain = simulate_db_run(i, response['recommendation'])
+        if i < first_n + 2:
+            average += gain / first_n
+        elif i > total_n - last_n + 2:
+            assert gain > average
+
+    average = 0
+    simulate_db_run(1, {'effective_cache_size': '0kB'})
+    for i in range(2, total_n + 2):
+        LOG.info('Test DNN (deep neural network)')
+        upload_result(result_dir='./integrationTests/data/', prefix='x__',
+                      upload_code='ottertuneTestTuningDNN')
+        response = get_result(upload_code='ottertuneTestTuningDNN')
+        assert response['status'] == 'good'
+        gain = simulate_db_run(i, response['recommendation'])
+        if i < first_n + 2:
+            average += gain / first_n
+        elif i > total_n - last_n + 2:
+            assert gain > average
+
+    average = 0
+    simulate_db_run(1, {'effective_cache_size': '0kB'})
+    for i in range(2, total_n + 2):
+        upload_result(result_dir='./integrationTests/data/', prefix='x__',
+                      upload_code='ottertuneTestTuningDDPG')
+        response = get_result(upload_code='ottertuneTestTuningDDPG')
+        assert response['status'] == 'good'
+        gain = simulate_db_run(i, response['recommendation'])
+        if i < first_n + 2:
+            average += gain / first_n
+        elif i > total_n - last_n + 2:
+            assert gain > average
```

Comment:
This assertion can catch the bug in algorithms with higher than 90% probability.

Reply:
You can also manually run 'fab integration_tests' and look at the graphs on the website to see if the trend is reasonable.